packages/wal/wal.go

```diff
@@ -50,9 +50,10 @@ func (w *WAL) NewChainWAL(chainID *iscp.ChainID) (chain.WAL, error) {
 	}
 	// read all segments in log
 	f, err := os.Open(w.dir)
-	defer f.Close()
 	if err != nil {
 		return nil, fmt.Errorf("could not open wal: %w", err)
+	} else {
+		defer f.Close()
 	}
```

Comment:
we can ditch the "elses"
```suggestion
	} 
	defer f.Close()
```

Reply:
I hate that. But that's how Go does it. Will change :-)